src/TimeModeling/Modeling/lsrtm_objective_parallel.jl

```diff
@@ -27,3 +27,16 @@ function lsrtm_objective(model::Model, source::judiVector, dObs::judiVector, dm;
     # first value corresponds to function value, the rest to the gradient
     return obj, gradient
 end
+
+function lsrtm_objective(model::Array{Model,1}, source::Array{judiVector{T,Array{T,2}},1}, dObs::Array{judiVector{T,Array{T,2}},1}, dm::Union{Array{Array{T,1},1}, Array{PhysicalParameter{T},1}}; options=Options(), nlind=false) where T
+# lsrtm_objective function for multiple sources and multiple vintages. The function distributes the sources and the input data amongst the available workers.
+
+    argout = judipmap((m, q, d, d_m) -> lsrtm_objective(m, q, d, d_m; options=options, nlind=nlind), model, source, dObs, dm)
+    results = [reduce((x, y) -> x .+ y, argout[i]) for i = 1:length(model)]
```

Comment:
Ok so last part is that currently you still gather abosultel;y everything then reduce while each model should do its own reduction. To do that t he reduction needs to be part of the base `judipmap` so that you have this multi-level reduction that is much more efficient.